### PR TITLE
Ignore .tito from source if we are not instructed to use it

### PR DESCRIPTION
**Pull request by jupierce**
Context: https://issues.redhat.com/browse/ART-1550 and specifically: https://issues.redhat.com/browse/ART-1551

### CODE REVIEW

File: doozerlib/rpmcfg.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 from __future__ import absolute_import, print_function, unicode_literals
 
 import glob
-import io
```
Comment (sosiouxme) [inline doozerlib/rpmcfg.py]: think ya still need this

File: doozerlib/rpmcfg.py [hunk 2]
```diff
@@ -140,16 +140,22 @@ def tito_setup(self):
...
-                exectools.cmd_assert('tito init')
-                # tito init will try to create a commit. We want this commit to be ignored
-                # from the perspective of any builds about to occur. So, we want to reset
-                # back to the original HEAD. The complexity here is that tito init will NOT
-                # make a commit if .gitignore is configured to somehow ignores the the .tito
-                # directory it creates. Thus, we store the SHA before running tito init
-                # and use that as a means to return to the upstream HEAD.
-                exectools.cmd_assert('git reset {}'.format(pre_init_sha))
+
+            # We either user .tito from source or we don't. If you want to use the upstream
```
Comment (sosiouxme) [inline doozerlib/rpmcfg.py]:
```suggestion
            # We either use .tito from source or we don't. If you want to use the upstream
```

File: doozerlib/rpmcfg.py [hunk 3]
```diff
@@ -140,16 +140,22 @@ def tito_setup(self):
...
+            # are going to clear it out and create our own.
+            if os.path.isdir(tito_dir):
+                shutil.rmtree(tito_dir)
+
+            pre_init_sha = exectools.cmd_assert('git rev-parse HEAD')[0].strip()
+            exectools.cmd_assert('tito init')
+            # tito init will try to create a commit. We want this commit to be ignored
+            # from the perspective of any builds about to occur. So, we want to reset
+            # back to the original HEAD. The complexity here is that tito init will NOT
+            # make a commit if .gitignore is configured to somehow ignores the the .tito
```
Comment (sosiouxme) [inline doozerlib/rpmcfg.py]:
```suggestion
            # make a commit if .gitignore is configured to somehow ignore the the .tito
```

Comment (sosiouxme) [reply doozerlib/rpmcfg.py]: honestly i would probably reduce this to "clear out anything tito may have added by reverting to previous HEAD"

Comment (jupierce) [reply doozerlib/rpmcfg.py]: Simplified comment.

File: doozerlib/rpmcfg.py (hunk 1 above)
Comment (jupierce) [reply doozerlib/rpmcfg.py]: `io` is still in there. I replaced a dupe import.

**Pull request merged**